.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,83 @@
+name: Test
+
+on:
+  pull_request:
+    branches: [ develop, beta, main ]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    name: Lint
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Read node version (package.json)
+        uses: culshaw/read-package-node-version-actions@v1
+        id: package-node-version
+
+      - name: Use Node.js (package.json)
+        uses: actions/setup-node@v1
+        with:
+          node-version: "${{ steps.package-node-version.outputs.version }}"
+
+      - name: Read `yarn cache dir`
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+
+      - name: Load yarn cache
+        uses: actions/cache@v2
+        id: yarn-cache
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Lint
+        uses: wearerequired/lint-action@v1.6.0
+        with:
+          github_token: ${{ secrets.github_token }}
+          eslint: true
+          eslint_args: "--max-warnings 0"
+          eslint_dir: src/
+
+  build:
```

Comment:
We can remove this job if you don't really care about having a test build in PRs. Yes this means all deployments are technically built twice. Consider it a dry run before the real thing happens :D